pandora/models/pandora.py

```diff
@@ -138,6 +92,76 @@ def get_audio_url(cls, data,
 
         return audio_url["audioUrl"] if audio_url is not None else None
 
+    def get_is_playable(self):
+        return self._api_client.transport.test_url(self.audio_url)
+
+
+class PlaylistItem(PlaylistModel):
+
+    artist_name = Field("artistName")
+    album_name = Field("albumName")
+    song_name = Field("songName")
+    song_rating = Field("songRating")
+    track_gain = Field("trackGain")
+    track_length = Field("trackLength", 0)
+    track_token = Field("trackToken")
+    audio_url = Field("audioUrl")
+    album_art_url = Field("albumArtUrl")
+    allow_feedback = Field("allowFeedback", True)
+    station_id = Field("stationId")
+
+    ad_token = Field("adToken")
```

Comment:
The way this is laid out it looks like it is possible to have a playlist item that is also an ad. When you look at a playlist item that is an ad do you then have to fetch the `AdItem` metadata to present the ad correctly or will you have it all within the `PlaylistItem`? In the later case it might make sense to combine the `AdItem` and `PlaylistItem` such that an ad is just another playlist item that the API consumer needs to call `register_ad` on.

If we don't get the full ad context back with the `get_playlist` call then we might want to look for ad items when we build the model list and build the appropriate models before we return the playlist the the consumer.